aptos-move/framework/aptos-token/sources/token_type.move

```diff
@@ -0,0 +1,162 @@
+// token_type provides a method to tell if a token is NFT, NFT print or a fungible token
+// NFT: only one token is allowed to mint from this TokenData. The maximum in TokenData is 1 and the field is immutable
+//
+// NFT_PRINT: this is a NFT with property_version > 0. Only one token has the same token_id
+// But, the maximum of the TokenData is not 1 or the maximum field is mutable at the same time.
+// Note: if token maximum is immutable and set to be 1, the token is still a NFT if property_version > 0
+//
+// FUNGIBLE: a non-decimal token that can have an amount bigger than 1
+module std::token_type {
```

Comment:
Do we really need this classification in the framework? The categories seem subjective to me and are up for interpretation, e.g. whether if token maximum is immutable => NFT vs NFT_PRINT. Can this just be an example (outside of the framework) and the exact interpretation should be up to NFT projects?